packages/drivers/odsp-driver/src/createFile/index.ts

```diff
@@ -15,15 +15,46 @@ export async function useCreateNewModule<T = void>(
 	// We can delay load this module as this path will not be executed in load flows and create flow
 	// while only happens once in lifetime of a document which happens in the background after creation of
 	// detached container.
-	const module = await import(/* webpackChunkName: "createNewModule" */ "./createNewModule.js")
-		.then((m) => {
-			odspLogger.sendTelemetryEvent({ eventName: "createNewModuleLoaded" });
-			return m;
-		})
-		.catch((error) => {
-			odspLogger.sendErrorEvent({ eventName: "createNewModuleLoadFailed" }, error);
-			throw error;
-		});
-
-	return func(module);
+
+	const maxRetries = 3;
+	const timeoutMs = 100; // 100ms
+	let lastError: Error;
+
+	for (let attempt = 1; attempt <= maxRetries; attempt++) {
+		try {
+			// Create timeout promise
+			const timeoutPromise = new Promise<never>((_resolve, reject) => {
+				setTimeout(
+					() => reject(new Error(`Import timed out after ${timeoutMs}ms`)),
+					timeoutMs,
+				);
+			});
+
+			// Race the import against the timeout
+			const module = await Promise.race([
```

Comment:
Why are you simultaneously awaiting the timeout and the import? Just wait when we fail to import. Just have a small 50ms timer.